### PR TITLE
feat: add euclidean_distance_transform op (#22038)

### LINKED ISSUE #22038: Add `euclidean_dist_transform`
This methods becomes very crucial in certian cases, i.e. loss calculation etc. 

- [Euclidean distance transform in tensorflow](https://stackoverflow.com/questions/47140842/euclidean-distance-transform-in-tensorflow)
- [tf-addons adaption](https://github.com/tensorflow/addons/blob/d208d752e98c310280938efa939117bf635a60a8/tensorflow_addons/image/distance_transform.py#L29)
- `from scipy.ndimage import distance_transform_edt`
- https://github.com/tensorflow/tensorflow/issues/24410

### PR DESCRIPTION
**Pull request by timon0305**
## Summary
- Add `euclidean_distance_transform` operation to compute the Euclidean distance from each foreground pixel to the nearest background pixel
- Implements support for TensorFlow, JAX, PyTorch, and NumPy backends using `scipy.ndimage.distance_transform_edt`
- OpenVINO backend raises `NotImplementedError`

## Test plan
- [x] Added unit tests for 2D, 3D, and 4D input shapes
- [x] Tests for sampling parameter
- [x] Tests for edge cases (all background, all foreground)
- [x] Verified against scipy reference implementation
- [x] Tests pass on TensorFlow, JAX, and Torch backends

Closes #22038

### CODE REVIEW

**Comment by jeffcarp** [comment]
Hi @timon0305 thanks for the submission, is this draft PR ready for review?
